platform/mbed_retarget.cpp

```diff
@@ -338,7 +337,10 @@ extern "C" int PREFIX(_write)(FILEHANDLE fh, const unsigned char *buffer, unsign
         n = length;
     } else {
         FileHandle* fhc = filehandles[fh-3];
-        if (fhc == NULL) return -1;
+        if (fhc == NULL) {
+            errno = EBADF;
```

Comment:
how breaking are this return codes? what is migration for existing applications?